Frameworks/Foundation/NSArray.mm

```diff
@@ -1041,32 +1028,43 @@ - (NSArray*)sortedArrayUsingFunction:(NSInteger (*)(id, id, void*))comparator co
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSUInteger)indexOfObjectIdenticalTo:(id)anObject inRange:(NSRange)range {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (range.location + range.length > [self count]) {
+        [NSException raise:NSRangeException
+                    format:@"Range {%d, %d} larger than array of size %d.", range.location, range.length, [self count]];
+    }
+
+    NSUInteger end = NSMaxRange(range);
+    for (NSUInteger i = range.location; i < end; ++i) {
+        if ([self objectAtIndex:i] == anObject) {
```

Comment:
should I make that part of these changes? I'm wary of changing too much in NSArray for possibly unnecessary optimization